File: api/python/examples/service.py
#encoding=utf8

from zbus import Service, ServiceConfig, ServiceHandler, Message, SingleBroker
import time

class MyServiceHandler(ServiceHandler):
    def handle_request(self, msg): 
        print msg
        res = Message()
        res.set_status('200')
        res.set_body('hello server@%s'%time.time())
        return res
 
config = ServiceConfig()

broker = SingleBroker(host='localhost', port=15555)
config.broker = broker
config.service_name = 'MyService'
config.thread_count = 1
config.service_andler = MyServiceHandler()


svc = Service(config)
svc.start()
svc.join()

broker.destroy()

File: NET/Zbus.Examples/Service.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;


using Zbus.Mq;
using Zbus.Net;
using Zbus.Broker;
using Zbus.Rpc;

namespace Zbus.Examples
{
    class ServiceTest
    {
        class MyServiceHandler : ServiceHandler
        {
            public Message HandleRequest(Message request)
            {
                request.SetBody("response from C#");
                return request;
            }
        }

        public static void Main(string[] args)
        {

            BrokerConfig config = new BrokerConfig();
            config.brokerAddress = "127.0.0.1:15555";
            Broker broker = new SingleBroker(config);

            ServiceConfig serviceConfig = new ServiceConfig(broker);
            serviceConfig.mq = "MyService";
            serviceConfig.serviceHandler = new MyServiceHandler();

            Service service = new Service(serviceConfig);
            service.Start();
        }
    }
}
